lib/web_ui/test/engine/semantics/semantics_test.dart

```diff
@@ -29,11 +29,32 @@ void main() {
 
 String rootSemanticStyle = '';
 
+/// Gets the root document or shadow root where the Flutter app is being rendered.
+///
+/// This function returns the correct root for the flutter app under testing,
+/// instead of hardcoding html.document across the test.
+///
+/// The root of a normal flutter app used to be html.document, but now that the
+/// whole app is wrapped in a Shadow DOM, that's not the case anymore.
+///
+/// A [html.ShadowRoot] quacks very similarly to a [html.Document], but unfortunately
+/// they don't share any class/implement any interface that let us use them interchangeably.
+/// To do so, just:
+///
+///   final root = getRootDocument();
+///
+/// So getRootDocument can be changed to return ShadowRoot or Document without
+/// the need to modify your code.
+///
+html.ShadowRoot getRootDocument() {
```

Comment:
I wouldn't call it "root document". It's not a document. Is more of a "host" for a Flutter view. I think a lot of these `querySelector`s can be cleaned up if we just had `DomRenderer` expose it directly.

Reply:
I tried to expose it directly through the DomRenderer, and it caused major breakage. I'll revisit :/

Reply:
I ended up calling this `appShadowRoot` and making it a getter, so I don't have to worry too much about the left-hand typing of it :)